src/Icons/src/Twig/IconFinder.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\Twig;
+
+use Symfony\Component\Finder\Finder;
+use Twig\Environment;
+use Twig\Loader\ChainLoader;
+use Twig\Loader\FilesystemLoader;
+use Twig\Loader\LoaderInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class IconFinder
+{
+    public function __construct(private Environment $twig)
+    {
+    }
+
+    /**
+     * @return string[]
+     */
+    public function icons(): array
+    {
+        $found = [];
+
+        foreach ($this->files($this->twig->getLoader()) as $file) {
+            $contents = file_get_contents($file);
+
+            if (preg_match_all('#ux_icon\(["\']([\w:-]+)["\']#', $contents, $matches)) {
```

Comment:
@smnandre, I'm hoping you can help refactor and improve this logic. 🙏🏻

Reply:
NodeVisitor :)) 

I'll look tuesday! 

We can find ideas in Translation extraction too i think